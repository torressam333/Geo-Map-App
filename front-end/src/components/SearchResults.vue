<template>
  <section class="absolute mt-2 w-full">
    <div
      v-if="searchQuery"
      class="h-[400px] overflow-scroll bg-white rounded-md"
    >
      <!-- Results API output -->
      <div
        class="px-4 py-2 flex gap-x-2 cursor-pointer hover:bg-slate-600 hover:text-white h-auto"
        v-for="result in searchData"
        :key="result.id"
      >
        <i class="fas fa-map-marker-alt"></i>
        <p class="text-xs" data-test="search-result">
          {{ result.place_name }}
        </p>
      </div>
    </div>
  </section>
</template>

<script>
export default {
  name: 'SearchResults',
  props: {
    searchData: {
      type: Object,
      required: false,
    },
    searchQuery: {
      type: String,
      required: false,
    },
  },
};
</script>
