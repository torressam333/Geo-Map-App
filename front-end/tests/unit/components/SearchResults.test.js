import { shallowMount } from '@vue/test-utils';
import SearchResults from '@/components/SearchResults';

describe('SearchResults', () => {
  const createResultsData = (searchDataResults = {}) => ({
    place_name:
      'Starbucks, 1580 N Verrado Way, Buckeye, Arizona 85396, United States',
    ...searchDataResults,
  });

  // Dynamic config object for wrapper mounting
  const createConfig = (searchDataResults) => ({
    propsData: {
      searchData: {
        ...searchDataResults,
      },
    },
  });

  it('has matching html output', () => {
    const wrapper = shallowMount(SearchResults);
    expect(wrapper.element).toMatchSnapshot();
  });

  it('renders an address', async () => {
    const searchDataResults = createResultsData([
      {
        place_name: 'Starbucks, 1580 fake street',
      },
    ]);

    const wrapper = shallowMount(
      SearchResults,
      createConfig(searchDataResults)
    );

    // Set the v-if to true
    await wrapper.setProps({ searchQuery: 'Starbucks' });

    expect(wrapper.text()).toMatch('Starbucks, 1580 fake street');
  });

  it('does not render results without a searchQuery', () => {
    const wrapper = shallowMount(SearchResults, createConfig());
    expect(wrapper.text()).toMatch('');
  });

  it('emits a method to pass selected location to parent', async () => {
    const wrapper = shallowMount(SearchResults);

    wrapper.vm.$emit('setResult');

    // Wait until $emits have been handled
    await wrapper.vm.$nextTick();

    expect(wrapper.emitted().setResult).toBeTruthy();
  });
});
