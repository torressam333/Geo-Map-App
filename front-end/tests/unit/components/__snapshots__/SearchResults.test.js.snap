// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResults has matching html output 1`] = `
<section
  class="absolute mt-2 w-full"
>
  <!--v-if-->
</section>
`;
